BeagleBone_black/src/sensors/vl6180.cpp

```diff
@@ -83,23 +85,34 @@ constexpr uint16_t INTERLEAVED_MODE__ENABLE              = 0x02A3;
 constexpr uint16_t RANGE_DEVICE_READY_MASK               = 0x01;
 constexpr uint16_t MODE_START_STOP                       = 0x01;
 constexpr uint16_t MODE_CONTINUOUS                       = 0x02;
+constexpr uint16_t MODE_SINGLESHOT                       = 0x00;
+constexpr uint16_t RESULT_INTERRUPT_STATUS_GPIO          = 0x4F;
+constexpr uint16_t INTERRUPT_CLEAR_RANGING               = 0x01;
+constexpr uint16_t RES_INT_RANGE_MASK                    = 0x07;
 
 namespace hyped {
 namespace sensors {
 
-Vl6180::Vl6180(uint8_t id, Logger& log)
-  :Thread(id, log)
+VL6180::VL6180(uint8_t i2c_addr, Logger& log)
+    : log_(log)
+    , on_(false)
+    , continuous_mode_(false)
 {
-  log_.INFO("VL6180", "Creating a sensor with id: %d", id);
+  // Create I2C instance get register address
+  this->i2c_addr_ = i2c_addr;
```

Comment:
`this` is not required, it can implicit in most cases,, use it only in case of name conflict

Reply:
👍 Also, there should never be name conflicts because of the trailing underscore